### PR TITLE
Update Quick Start Guide

### PR DESCRIPTION
**Pull request by Malax**
As pointed out in https://github.com/heroku/libcnb.rs/pull/583#discussion_r1297831464, the output of `cargo libcnb package` and `pack build` has changed since we last updated the guide. This PR brings the output and output paths up-to-date.

Ref: GUS-W-13972958

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,16 +178,22 @@ In your project directory, run `cargo libcnb package` to start packaging:
...
+Writing buildpack directory...
+Successfully wrote buildpack directory: packaged/x86_64-unknown-linux-musl/debug/libcnb-examples_my-buildpack (4.06 MiB)
+✨ Packaging successfully finished!
+
+💡 To test your buildpack locally with pack, run:
+pack build my-image-name \
+  --buildpack /home/ponda.baba/my-buildpack/packaged/x86_64-unknown-linux-musl/debug/libcnb-examples_my-buildpack \
+  --path /path/to/application
+
+/home/ponda.baba/my-buildpack/packaged/x86_64-unknown-linux-musl/debug/libcnb-examples_my-buildpack
```
Comment (edmorley) [inline README.md:196]: (Fixing these is covered by #599 and #600 btw)

File: README.md [hunk 2]
```diff
@@ -178,16 +178,22 @@ In your project directory, run `cargo libcnb package` to start packaging:
...
+Building binaries (x86_64-unknown-linux-musl)...
 # Omitting compilation output...
-    Finished dev [unoptimized] target(s) in 8.51s
-INFO - Writing buildpack directory...
-INFO - Successfully wrote buildpack directory: target/buildpack/debug/libcnb-examples_my-buildpack (3.03 MiB)
-INFO - Packaging successfully finished!
-INFO - Hint: To test your buildpack locally with pack, run: pack build my-image --buildpack target/buildpack/debug/libcnb-examples_my-buildpack --path /path/to/application
+    Finished dev [unoptimized] target(s) in 8.92s
+Writing buildpack directory...
+Successfully wrote buildpack directory: packaged/x86_64-unknown-linux-musl/debug/libcnb-examples_my-buildpack (4.06 MiB)
```
Comment (edmorley) [inline README.md:188]: I'm curious why the buildpack size has increased by 30% since the last time this was refreshed?

**Review by edmorley — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Malax) [reply README.md:188]: I saw that too and knew you would latch onto it immediately :) It could really be anything since this is a debug build - rustc might have added more debug symbols or, more likely, we added more dependencies since we wrote the guide. I think I remember writing it before most of us even had CNBs for their languages.

FWIW: I re-ran this with `--release` and the size dropped to 1.78 MiB.

File: README.md (hunk 1 above)
Comment (Malax) [reply README.md:196]: Ah nice, I was annoyed when it had my name in the output. OTOH, adding Star Wars references is also great. ;-) We need to make sure to do another update of the guide when #599 and #600 get merged then.

File: README.md (hunk 2 above)
Comment (edmorley) [reply README.md:188]: The guide was last refreshed for libcnb 0.11.2, so not that long ago. Agree it might be debug-only inflation, since some new transitive but not really used dep might not get pruned in debug mode. Looking at the diff there were also some `toml` and `fancy-regex` major version upgrades - I know both projects have had significant overhauls (eg toml had a refactor due to wanting to support editing too etc).

Comment (edmorley) [reply README.md:188]: I saw increases in Procfile CNB release builds too - which I looked into here:
https://github.com/heroku/procfile-cnb/pull/175#issuecomment-1686312268

tl;dr: Mostly regex + toml dependency changes. We should also stop using regex when we don't need to use regex.
